File: leetcode/go/219.contains-duplicate-ii.go
func containsNearbyDuplicate(nums []int, k int) bool {
    m := make(map[int]int, len(nums))
    for i, n := range nums {
        if val, ok := m[n]; ok && i - val <= k {
            return true
        }
        m[n] = i
    }
    return false
}

File: leetcode/rust-leetcode/src/219.contains-duplicate-ii.rs
impl Solution {
    pub fn contains_nearby_duplicate(nums: Vec<i32>, k: i32) -> bool {
        let mut map = std::collections::HashMap::new();
        let k = k as usize;
        for i in 0..nums.len() {
            if map.contains_key(&nums[i]) && i - map.get(&nums[i]).unwrap() <= k {
                return true;
            }
            map.insert(nums[i], i);
        }
        false
    }
}

